controllers/ocsclient_controller.go

```diff
@@ -0,0 +1,475 @@
+		if errCode == codes.InvalidArgument {
+			msg = "StorageConsumer UID is not valid. Contact the provider admin"
+			//eventReason = "UIDInvalid"
+			//eventType = corev1.EventTypeWarning
+		}
+	} else if grpcCallName == UpdateCapacity {
+		if errCode == codes.InvalidArgument {
+			msg = "StorageConsumer UID or requested capacity is not valid. Contact the provider admin"
+			//eventReason = "UIDorCapacityInvalid"
+			//eventType = corev1.EventTypeWarning
+		} else if errCode == codes.NotFound {
+			msg = "StorageConsumer UID not found. Contact the provider admin"
+			//eventReason = "UIDNotFound"
+			//eventType = corev1.EventTypeWarning
+		}
+	} else if grpcCallName == GetStorageConfig {
+		if errCode == codes.InvalidArgument {
+			msg = "StorageConsumer UID is not valid. Contact the provider admin"
+			//eventReason = "UIDInvalid"
+			//eventType = corev1.EventTypeWarning
+		} else if errCode == codes.NotFound {
+			msg = "StorageConsumer UID not found. Contact the provider admin"
+			//eventReason = "UIDNotFound"
+			//eventType = corev1.EventTypeWarning
+		} else if errCode == codes.Unavailable {
+			msg = "StorageConsumer is not ready yet. Will requeue after 5 second"
+			//eventReason = "NotReady"
+			//eventType = corev1.EventTypeNormal
+		}
+	}
+
+	if msg != "" {
+		r.Log.Error(err, "StorageProvider:"+grpcCallName+":"+msg)
+		// r.recorder.ReportIfNotPresent(instance, eventType, eventReason, msg)
+	}
+}
+
+func (r *OcsClientReconciler) createAndOwnStorageClassClaim(instance *v1alpha1.OcsClient, claim *v1alpha1.StorageClassClaim) error {
+
+	err := controllerutil.SetOwnerReference(instance, claim, r.Client.Scheme())
+	if err != nil {
+		return err
+	}
+
+	claim.Annotations = map[string]string{
+		ocsClientAnnotation: fmt.Sprintf("%s/%s", instance.Namespace, instance.Name),
+	}
+
+	err = r.Client.Create(context.TODO(), claim)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+	return nil
+}
+
+// claims should be created only once and should not be created/updated again if user deletes/update it.
+func (r *OcsClientReconciler) createDefaultStorageClassClaims(instance *v1alpha1.OcsClient) error {
+
+	storageClassClaimFile := &v1alpha1.StorageClassClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateNameForCephFilesystemSC(instance.Name),
+			Namespace: instance.Namespace,
+			Labels:    map[string]string{
+				//defaultStorageClassClaimLabel: "true",
+			},
+		},
+		Spec: v1alpha1.StorageClassClaimSpec{
+			Type: "sharedfilesystem",
+		},
+	}
+
+	storageClassClaimBlock := &v1alpha1.StorageClassClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateNameForCephBlockPoolSC(instance.Name),
+			Namespace: instance.Namespace,
+			Labels:    map[string]string{
+				//defaultStorageClassClaimLabel: "true",
+			},
+		},
+		Spec: v1alpha1.StorageClassClaimSpec{
+			Type: "blockpool",
+		},
+	}
+
+	err := r.createAndOwnStorageClassClaim(instance, storageClassClaimFile)
```

Comment:
Move this to line 450 so that struct and function will remain next to reach other.